src/libraries/System.Private.CoreLib/src/System/BitConverter.cs

```diff
@@ -767,124 +767,84 @@ public static bool ToBoolean(ReadOnlySpan<byte> value)
         /// </summary>
         /// <param name="value">The number to convert.</param>
         /// <returns>A 64-bit signed integer whose bits are identical to <paramref name="value"/>.</returns>
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static unsafe long DoubleToInt64Bits(double value)
-        {
-            // Workaround for https://github.com/dotnet/runtime/issues/11413
-            if (Sse2.X64.IsSupported)
-            {
-                Vector128<long> vec = Vector128.CreateScalarUnsafe(value).AsInt64();
-                return Sse2.X64.ConvertToInt64(vec);
-            }
-
-            return *((long*)&value);
-        }
+        [Intrinsic]
+        public static unsafe long DoubleToInt64Bits(double value) => *((long*)&value);
```

Comment:
Does the removal of the SSE2 path have any negative impact when using mono?

Reply:
Possibly for Mono JIT. Mono LLVM correctly handles this as a bitcast already.

Reply:
Mono JIT doesn't support those at all. LLVM will be 100% fine without it 🙂